light/provider/http/http.go

```diff
@@ -18,20 +18,28 @@ var (
 	// This is very brittle, see: https://github.com/tendermint/tendermint/issues/4740
 	regexpMissingHeight = regexp.MustCompile(`height \d+ (must be less than or equal to|is not available)`)
 
-	maxRetryAttempts = 10
-	timeout          = 5 * time.Second
+	defaultMaxRetryAttempts = 10
+	defaultTimeout          = 5 * time.Second
 )
 
 // http provider uses an RPC client to obtain the necessary information.
 type http struct {
 	chainID string
 	client  rpcclient.RemoteClient
+
+	maxRetryAttempts int
 }
 
 // New creates a HTTP provider, which is using the rpchttp.HTTP client under
 // the hood. If no scheme is provided in the remote URL, http will be used by
 // default. The 5s timeout is used for all requests.
 func New(chainID, remote string) (provider.Provider, error) {
+	return NewWithOptions(chainID, remote, defaultMaxRetryAttempts, defaultTimeout)
+}
+
+// NewWithOptions is an extension to creating a new http provider that allows the addition
+// of a specified timeout and maxRetryAttempts
+func NewWithOptions(chainID, remote string, maxRetryAttempts int, timeout time.Duration) (provider.Provider, error) {
```

Comment:
Could we maybe group options into struct

```
struct Options {
  // -1 means no limit
  MaxRetryAttempts int
  // 0 means no timeout.
  Timeout time.Duration
}
```

or https://www.sohamkamani.com/golang/options-pattern/

Reply:
This would mean that adding options wouldn't be a breaking change right?

I will make a new PR for this